bare_metal_tests/TESTS/bringup/uart/test.py

```diff
@@ -0,0 +1,330 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import time
+import serial
+import sys
+import mbed_lstools
+from subprocess import Popen, PIPE, STDOUT
+import shutil
+import argparse
+import ntpath
+import json
+import os
+import logging
+
+"""! UART TEST FOR GREEN TEA
+
+  TEST SCENARIO
+  This part of the test is executed on the host (PC).
+ 
+  After startup it will try to communicate via UART with the mbed device.
+  This part of the test waits for the expected message and responds with
+  the same message. 
+  4 test messages should be received. Last message ("PASSED") should inform
+  that on the mbed side also all messages have been received.
+ 
+  To perform this test with serial port baud rate different than default (9600),
+  call host test with appropriate --baudrate parameter.
+  
+  Example usage:
+      test.py -m K64F -t GCC_ARM -b 9600
+  This command will run test on K64F platfom using GCC_ARM toolchain and
+  serial connection speed equal to 9600 b/s.
+  
+  For more options:
+      test.py -h
+      
+"""
+
+serial_port = None
+RETRY_LIMIT = 10
+
+# List of expected messages from the mbed device.
+message_list = ['x\0', 
+                'mbed\0',
+                'Green Tea UART bear metal test.\0', 
+                'Long Message-Long Message-Long Message-Long Message-Long Message-Long Message-Long Message-Long Message-Long Message--.\0',
+                'PASSED\0']
+
+def exit():
+    if(serial_port != None and serial_port.isOpen() == True):
+        serial_port.close()
+    sys.exit()
+        
+def create_test_header_file(baudrate):
+    """! Create header file with the specified baud rate.
+    
+    @param baudrate specified transmission baud rate.
+    """
+    file = open('test_config.h','w') 
+    file.write('/* This file has been automatically generated by the python script. */\n')
+    file.write('/* This file contains serial port baud rate. */\n')
+    file.write('#define BAUD_RATE %s\n' % baudrate)
+    file.close()
+
+def run_command(cmd):
+    """! Runs command and prints proc stdout on screen.
+         It is used to build the test.
+         
+    @param cmd command to be executed.
+    @return image path.
+    """
+    image_path = str()
+    
+    # Remove build log file if exists
+    try:
+        os.remove('build.log')
+    except OSError:
+        pass
+    
+    try:
+        p = Popen(cmd, stdout=sys.stdout)
+    except OSError as e:
+        logging.error('Command can not be executed.')
+        exit()
+
+    # Check process status.
+    returncode = p.wait()
+    if returncode < 0:
+        logging.error('Error while executing command.')
+        exit()
+    
+    # Get image path     
+    try:
+        with open('build.log') as f:
+            print('file ok')
+            build_data = json.load(f)
+            print('json ok')
+            image_path = build_data['builds'][0]['bin']
+            print('json ok')
+    except:
```

Comment:
Could you narrow the exceptions you catch here? As it stands it would catch things like `C-c` and out of memory conditions.

Reply:
Fixed.